cmd/kyma/undeploy/cmd.go

```diff
@@ -84,7 +84,7 @@ func (cmd *command) Run() error {
 	}
 
 	if !cmd.opts.NonInteractive {
-		if err := cli.DetectManagedEnvironment(cmd.K8s, cmd.Factory.NewStep("")); err != nil {
+		if err := cli.DetectManagedEnvironment(context.Background(), cmd.K8s, cmd.Factory.NewStep("")); err != nil {
```

Comment:
```suggestion
		if err := cli.DetectManagedEnvironment(ctx, cmd.K8s, cmd.Factory.NewStep("")); err != nil {
```
you actually can use the command contact if you adjust the command Run above like this


```golang

	cobraCmd := &cobra.Command{
		Use:   "undeploy",
		Short: "Undeploys Kyma from a running Kubernetes cluster.",
		Long:  `Use this command to undeploy Kyma from a running Kubernetes cluster.`,
		RunE:  func(cobraCmd *cobra.Command, _ []string) error { return cmd.Run(cobraCmd.Context()) },
	}

```

and then pass it with

```golang
func (cmd *command) Run(ctx context.Context) error {
```

Reply:
:+1: